lib/socketServer.js

```diff
@@ -38,10 +30,29 @@ io.on('connection', ({ socket, state }) => {
 })
 
 module.exports = (app, { verbose }) => client => {
-  if (verbose) io.use(log())
+  if (verbose) {
+    io.use(async (context, next) => {
+      console.log(`<- ${context.event} ${context.data}`)
+      await next()
+      console.log(`<- ${context.event} ${context.data}`)
+    })
+  }
 
   const subscriptions = new Set()
-  io.state = { client, subscriptions }
+
+  io.use(async (context, next) => {
+    context.client = client
+    context.subscriptions = subscriptions
+    context.ad = ad
+    await next()
+  })
+
   io.attach(app)
+
+  // FIXME: wrap app.listen to do this
+  const name = process.env.LIVING_ROOM_NAME || require('os').hostname()
+  const bonjour = require('nbonjour').create()
+  bonjour.publish({name, type: 'socketio', protocol: 'tcp', subtypes: ['livingroom'], port: 3000})
```

Comment:
Unfortunately I couldnt figure out how to do `app.on('listening', () => bonjour.publish...`.

Reply:
Bummer :)

Reply:
Im figuring out a workaround now. I'm gonna wrap each server into something called a Service that will call the listen and emit correctly.